### PR TITLE
Initial PJS protos

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR contains the initial PJS protos discussed here: https://www.notion.so/2023-10-03-Pachyderm-Job-System-1a6aca512d494adc837e4632d2127576.

### CODE REVIEW

File: src/pjs/pjs.proto
Comment (jrockway) [reply src/pjs/pjs.proto:4]: Yeah.  pjs.pb.go isn't even checked in, it appears.  There is something I want to look at in the generated files, so we should fix that.

**Review by brendoncarroll — APPROVED** [review]

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:14]: I don't think these enums need the type name in the value names.   The generated code calls it `JobState_JOB_STATE_UNSPECIFIED`, so it just stutters.

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:36]: As mentioned above, we should decide what the "forgot to set it" state is; nil job, or job.job_id = 0, but not both.

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:10]: I feel like embedded messages with one field have caused a lot of pain over the years.  They provide 2 different invalid states; nil Job, and non-nil Job but id = 0.  We should only have one state for unknown/unset job ids.  In that case, I feel like having `int64 job_id` in each message that references a job would be adequate; 0 indicates "the caller forgot to set it".

If we want to keep the embedded message, we should decide right now what the "unset" state is.  Either this should have a validation rule that id is not equal to 0, or each field that takes a job_id as input should have a Job != nil validation rule (`[(validate.rules).message.required = true];`).

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:37]: I assume that there can be parentless jobs, so this is probably a case where a nil Job is the right identifier.  But we should be sure to reject a zero-valued job in that case.

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:35]: Let's document the messages and the fields.

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:90]: I think we should avoid this pattern of RPCs returning system objects; ListJob should return ListJobResponse; if for now that only contains a JobInfo, that's fine, but the hook for future extensibility without affecting other uses of JobInfo is always appreciated.

File: src/pjs/pjs.proto
Comment (jrockway) [inline src/pjs/pjs.proto:119]: Even after reading the doc, I'm not sure what a context is.  It says it identifies the parent of this job, but parents of jobs are identified as Job objects elsewhere.  Why a string and how does a caller produce a correct string?

File: src/typescript/pjs/pjs.pb.ts
Comment (jrockway) [inline src/typescript/pjs/pjs.pb.ts:1]: Someone should edit .gitattributes to make src/typescript/*/*.pb.ts as generated files.

**Review by jrockway — APPROVED** [review]

File: src/pjs/pjs.proto
Comment (brendoncarroll) [reply src/pjs/pjs.proto:119]: This is a bearer token used to create a job from within another job.

File: src/pjs/pjs.proto
Comment (brendoncarroll) [reply src/pjs/pjs.proto:14]: I don't think you can have two enums with the same value in the same package?  Something about c++.

File: src/pjs/pjs.proto
Comment (bbonenfant) [reply src/pjs/pjs.proto:119]: Could this info be included as comment(s) in this file

**Review by bbonenfant — APPROVED** [review]
No need to hold this up, but it helps downstream clients a lot when all additions to the proto file are documented with comments. All the RPCs have great documentation, but the messages at the bottoms of pjs.proto file are a little bare.

File: src/pjs/pjs.proto
Comment (jrockway) [reply src/pjs/pjs.proto:14]: That seems to be true.

**Pull request merged**
